toolchain/lowering/lowering_function_context.h

```diff
@@ -93,6 +93,19 @@ class FunctionContext {
     return synthetic_block_ == block;
   }
 
+  // After emitting an initializer `init_id`, finishes performing the
+  // initialization of `dest_id` from that initializer. This is a no-op if the
+  // initialization was performed in-place, and otherwise performs a store or a
+  // copy.
+  auto FinishInitialization(SemIR::TypeId type_id, SemIR::NodeId dest_id,
+                            SemIR::NodeId init_id) -> void;
+
+  // Emit a value copy for type `type_id` from `source_id` to `dest_id`.
+  // `source_id` must produce a value representation for `type_id`, and
+  // `dest_id` must be a pointer to a `type_id` object.
+  auto CopyValue(SemIR::TypeId type_id, SemIR::NodeId source_id,
+                 SemIR::NodeId dest_id) -> void;
```

Comment:
Should this be private? It looks like it might just be an implementation detail of FinishInit, although maybe that's temporary.

Reply:
Done. (If we end up wanting to use it more generally it's trivial to make it public again.)